### PR TITLE
Add kitgen README 

### PR DESCRIPTION
**Pull request by ermik**
This measly PR with absolutely no code change and no performance improvements seeks to address #639 with a bit of note-taking.

/request @nyarly

### CODE REVIEW

**Comment by ermik** [comment]
👇🏻haven't touched those failing tests.

File: cmd/kitgen/README.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+Service interface and any of the custom types it refers to:
+```go
+// service.go
+package profilesvc // don't forget to name your package
+
+type Service interface {
+    PostProfile(ctx context.Context, p Profile) error
+    // ...
+}
+type Profile struct {
```
[the remark this answers is not in the document]
Comment (ermik) [reply cmd/kitgen/README.md]: @nyarly — found what broke the codegen. See #745

**Comment by peterbourgon** [comment]
This seems reasonable to me but I'll wait for a plus-one from @nyarly

**Comment by ermik** [comment]
@peterbourgon perhaps add him as a reviewer?

**Comment by ermik** [comment]
@nyarly if you want a tester — I am running kitgen almost daily atm.

**Comment by peterbourgon** [comment]
Thanks!

**Pull request merged**